File: 824-number-of-lines-to-write-string/number-of-lines-to-write-string.php
class Solution {

    /**
     * @param Integer[] $widths
     * @param String $s
     * @return Integer[]
     */
    function numberOfLines($widths, $s) {
        $r=1;
        $c=0;
        foreach(str_split($s) as $ch){
            $width=$widths[ord($ch)-97];
            $r=$c + $width > 100 ? $r + 1 : $r;
            $c=$c + $width > 100 ? $width:$width+$c;
        }
        return [$r,$c];
    }
}

File: 824-number-of-lines-to-write-string/number-of-lines-to-write-string.cs
public class Solution {
    public int[] NumberOfLines(int[] widths, string s) {
        int r=1,c=0;
        foreach(char ch in s){
            int width=widths[ch-'a'];
            r=width+c>100?r+1:r;
            c=width+c>100?width:width+c;
        }
        return new int[]{r,c};
    }
}
